fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -751,7 +762,7 @@ private ScoredPlan planIndex(@Nonnull PlanContext planContext, @Nonnull QueryCom
                 final List<QueryComponent> unsatisfiedFilters = filter instanceof AndComponent ?
                                                                 ((AndComponent) filter).getChildren() :
                                                                 Collections.singletonList(filter);
-                p = new ScoredPlan(0, p.plan, unsatisfiedFilters, p.createsDuplicates);
+                p = new ScoredPlan(0, p.info, unsatisfiedFilters, p.createsDuplicates, p.isStrictlySorted);
```

Comment:
Note to the reviewer: I am inheriting p.isStrictlySorted which previously was set to false.